docs/markdown/snippets/rust-mixed.md

```diff
@@ -0,0 +1,4 @@
+## Rust and non-Rust sources in the same target
+
+Meson now supports creating a single target with Rust and non Rust
+sources mixed together, as long as the `link_language` is `rust`.
```

Comment:
```suggestion
sources mixed together. In this case, if specified, `link_language` must be set to `rust`.
```
or something similar: IIUC the implementation, if `link_language` is not set, it is set to `rust` automatically.